src/test/java/api/requests/PickSlipsTests.java

```diff
@@ -35,326 +40,356 @@
+      .page()
+      .withPickupServicePointId(circDesk1)
+      .forItem(planetSecondFloorCd4)
+      .by(jessica));
+
+    // response for Circ Desk 1
     Response responseForCd1 = ResourceClient.forPickSlips().getById(circDesk1);
     assertThat(responseForCd1.getStatusCode(), is(HTTP_OK));
-
     assertResponseHasItems(responseForCd1, 1);
-    validateResponse(responseForCd1, planetThirdFloorCd1);
-    assertThat(responseForCd1.getJson().getJsonArray(PICK_SLIPS_KEY).getJsonObject(0).getString(ID_KEY),
-        is(planetThirdFloorCd1.getId().toString()));
+    assertResponseContains(responseForCd1, planetThirdFloorCd1, requestForThirdFloorCd1, charlotte);
 
-    // Report for Circ Desk 4
-    UUID circDesk4 = servicePointsFixture.cd4().getId();
+    // response for Circ Desk 4
     Response responseForCd4 = ResourceClient.forPickSlips().getById(circDesk4);
     assertThat(responseForCd4.getStatusCode(), is(HTTP_OK));
-
     assertResponseHasItems(responseForCd4, 1);
-    validateResponse(responseForCd4, planetSecondFloorCd4);
-    assertThat(responseForCd4.getJson().getJsonArray(PICK_SLIPS_KEY).getJsonObject(0).getString(ID_KEY),
-        is(planetSecondFloorCd4.getId().toString()));
+    assertResponseContains(responseForCd4, planetSecondFloorCd4, requestForSecondFloorCd4, jessica);
   }
 
-  private void validateResponse(Response response, InventoryItemResource... sourceItems) {
-    Stream.of(sourceItems).forEach(sourceItem -> {
-      Optional<JsonObject> matchingItemFromResponse = response
-          .getJson()
-          .getJsonArray(PICK_SLIPS_KEY).stream()
-          .map(JsonObject.class::cast)
-          .filter(itemFromResponse -> sourceItem.getId().toString().equals(itemFromResponse.getString(ID_KEY)))
-          .findFirst();
-
-      if (matchingItemFromResponse.isPresent()) {
-        validateItem(matchingItemFromResponse.get(), sourceItem);
-      } else {
-        fail("Expected item is missing in response: itemId=" + sourceItem.getId());
-      }
-    });
+  private void assertResponseHasItems(Response response, int itemsCount) {
+    JsonObject responseJson = response.getJson();
+    assertThat(responseJson.getJsonArray(PICK_SLIPS_KEY).size(), is(itemsCount));
+    assertThat(responseJson.getInteger(TOTAL_RECORDS), is(itemsCount));
   }
 
-  private void validateItem(JsonObject item, InventoryItemResource sourceItem) {
-    assertThat(item.getString(ID_KEY), is(sourceItem.getId().toString()));
-    assertThat(item.getString(INSTANCE_ID_KEY), is(sourceItem.getInstanceId().toString()));
-    assertThat(item.getString(HOLDINGS_RECORD_ID_KEY), is(sourceItem.getHoldingsRecordId().toString()));
-    assertThat(item.getString(BARCODE_KEY), is(sourceItem.getBarcode()));
-    assertThat(item.getString(TITLE_KEY), is(sourceItem.getInstance().getResponse().getJson().getString(TITLE_KEY)));
+  private void assertResponseContains(Response response, InventoryItemResource item, IndividualResource request, IndividualResource requester) {
```

Comment:
Please can the method parameters be split onto a second line to meet FOLIO's line length limits

Reply:
fixed